src/ants_gbt.erl

```diff
@@ -0,0 +1,107 @@
+%%%-------------------------------------------------------------------
+%%% @author Daniel Grzonka
+%%% @copyright (C) 2014, <COMPANY>
+%%% @doc
+%%% Ants gb_trees based approach
+%%% @end
+%%% Created : 07. lis 2014 15:56
+%%%-------------------------------------------------------------------
+-module(ants_gbt).
+-author("Daniel").
+
+%% API
+-export([create_ants/4, partition/3, get_agent/3, update_agent/4, group_by/1]).
+-type ant_state() :: parallant:ant_state().
+
+-include("parallant.hrl").
+
+-spec create_ants(PopulationSize :: pos_integer(),
+                  Width :: dimension(),
+                  Height :: dimension(),
+                  config()) ->
+                         Ants :: gb_trees:tree().
+create_ants(PopulationSize, Width, Height, Config) ->
+    Pop = model:initial_population(PopulationSize, Width, Height, Config),
+    IndividualsWithKeys = [{Pos,
+                            #ant{pos = Pos, state = State}}
+                           || {Pos, State} <- Pop],
+    gb_trees:from_orddict(IndividualsWithKeys).
+
+%% -spec get_agent(position(), environment(), config()) -> ant_state().
+%% get_agent(Position, Env, _Config) ->
+%% case gb_trees:lookup(Position,  Env#env.agents) of
+%%   {value, Value} ->
+%%     #ant{state = State} = Value,
+%%     State;
+%%   none ->
+%%     empty
+%% end.
+
+-spec get_agent(position(), environment(), config()) -> ant_state().
+get_agent(Position, Env, _Config) ->
+    io:format("~n***********~n"),
+    io:format("Position: ~p~n", [Position]),
+    io:format("Agents: ~p~n", [gb_trees:values(Env#env.agents)]),
+    io:format("AgentsTree: ~p~n", [Env#env.agents]),
+    Filtered = [State || #ant{pos = Pos, state = State} <- gb_trees:values(Env#env.agents),
```

Comment:
Line 46 is too long: "    Filtered = [State || #ant{pos = Pos, state = State} <- gb_trees:values(Env#env.agents),".